_posts/2019-12-18-sentry-the-guardian-of-your-quarkus-app.adoc

```diff
@@ -0,0 +1,80 @@
+. *You were waiting for this article 🤓*
+
+Now that you are teased, aren't you? Let's dive in. 
+
+== Sentry, the guardian has arrived on Quarkus
+
+Once your code is reviewed, tested in many ways, containerized, ..., and finally *promoted to production*, [line-through]#I guess you can expect it to run smoothly ever after..# *an exception is going to pop*, it's just a question of time. The questions are:
+
+. *How will you get notified about this error?*
+. *What are you going to do about it?*
+
+Well, this looks like a complex problem right? This is what I really like about using *Quarkus and Sentry together*, you will be all set in a few minutes:
+
+* Create a https://sentry.io/signup/[Sentry account], you can choose a free plan or another depending on your project needs (or with a bit more time, even a https://docs.sentry.io/server/[self hosted sentry])
+* Configure Sentry on your Quarkus application
+. Add the extension:
++
+```
+$ ./mvnw quarkus:add-extension -Dextensions=sentry
+```
+. Set it up: 
++
+```
+quarkus.log.sentry=true
+quarkus.log.sentry.dsn=[your Sentry DSN]
+quarkus.log.sentry.in-app-packages=org.acme
+```
++ 
+NOTE: More details about configuration can be found in the https://quarkus.io/guides/logging-sentry[guide]
+
+== Happily ever after
+
+You are now pretty confident that in the very unlikely event of an error in your fully tested code, *your guardian will be watching*..
+
+*Bip bip*, you've got mail.. ewww snap an error 🥳 (and you are happy that you haven't missed it)
+
+image::sentry-email.png[Sentry Email Notification, 900]
+
+You now enter the really pretty Sentry application. It is also smart enough to *group similar events into issues* (or you can also decide to group them yourself). 
+
+image::sentry-issues.png[Sentry Issue List, 900]
+
+It's now time to *generate and link an issue on your favorite tracker* (GitHub, Bitbucket, GitLab, ...). It will contain a lot of *contextual information about the errors* (endpoint, params, stacktrace, environment, ...) to help your team fix that nasty bug. 
+
+image::sentry-issue.png[Sentry issue details, 900]
+
+If you want to go a bit further you can start cooking some *custom rules to dispatch alerts* with your own taste.
+
+image::sentry-alerts-config.png[Sentry Alerts Configuration, 900]
+
+Bonus point, *Sentry is also available for your https://sentry.io/for/javascript/[frontend]*.
+
+Good luck 🤙
+
+NOTE: Once an extension is integrated in the Quarkus eco-system, the life of thousand of developers is made easier. If you know of a similar (or any) tool you think could help the Quarkus community, https://github.com/quarkusio/quarkus/issues/new?assignees=&labels=extension-proposal&template=extension_proposal.md&title=[propose it]!
+
+NOTE: Sentry has recently https://blog.sentry.io/2019/11/06/relicensing-sentry[re-licensed most of its code to BSL] - which is not an OSI-approved license. The Java SDK is still under the 3-Clause BSD license.
```

Comment:
What's the impsct this has in practical terms?

Reply:
It's the new "we don't like Amazon using our stuff without paying" type of license.

Basically:
- you have access to the source code and can modify it but there are restrictions about the usage - typically you may have to pay to use it in production under certain conditions
- the software will eventually become Open Source after a period of time (counted in years)

So that's definitely not an Open Source software anymore. And this is going to be a major pain.

Reply:
IANAL, but yes, and it means a company like red hat which easily can be categorised as a company doing services that help trace errors upgrades to use latest sentry code on their own-hosted services they are in breach of license.  

in other words - to continue use sentry from our opensource projects using own hosted software we'll most likely have to use pre-november sourcecode hosted sentry.

Reply:
Also, IMO that NOTE should be a WARNING and I would prefer it to be higher up towards the section on how to setup sentry account.